include/links-model.php

```diff
@@ -48,9 +48,24 @@ public function __construct( &$model ) {
 
 		$this->home = home_url();
 
-		// Hooked with normal priority because it needs to be run after static pages is set in language data.
+		// Hooked with normal priority because it needs to be run after static pages is set in language data. Must be done early (before languages objects are created).
```

Comment:
This additional comment makes me fear that moving the other filters to `init()` is also a bad idea. Sorry for the wrong path I pushed you.